src/app/modules/browse/browse-controller.js

```diff
@@ -140,9 +140,29 @@
             }
             return _localize(message, params);
         }
+        function setHiddenBooksMessages() {
+            if ($scope.allLicenses) {
```

Comment:
This whole thing is not great for localization because it assumes that the part of "Hide books with..." that should be the link will always come first, and the bit of "Hiding... Show them" will always come last. What if the only way to express it well in some language is something like "5 of these books have restricted licenses [[Hide them]]" and "[[Show]] the 5 books that are hidden because..."?
What if we made a single localize string for each case, like "[Hide] books with restricted/unknown licenses." and "Hiding {count} books with restricted/unknown licences. [Show them]". Then in both cases you could replace {count}, if present, with the count, and extract the bit before [ to hiddenBooksMessage, the bit inside to the toggle bit,  and the bit after the ] to closing.
If you still want to do a special case for count 1 that would be another single string, which could also have "show it".